client/checkout/classic/index.js

```diff
@@ -0,0 +1,368 @@
+		$form.removeClass( 'processing' ).submit();
+	};
+
+	// Create payment method on submission.
+	let paymentMethodGenerated;
+
+	/**
+	 * Generates a payment method, saves its ID in a hidden input, and re-submits the form.
+	 *
+	 * @param {object} $form The jQuery object for the form.
+	 * @param {function} successHandler    Callback to be executed when payment method is generated.
+	 * @param {boolean}  useBillingDetails Flag to control whether to use from billing details or not.
+	 * @return {boolean} A flag for the event handler.
+	 */
+	const handlePaymentMethodCreation = (
+		$form,
+		successHandler,
+		useBillingDetails
+	) => {
+		// We'll resubmit the form after populating our payment method, so if this is the second time this event
+		// is firing we should let the form submission happen.
+		if ( paymentMethodGenerated ) {
+			paymentMethodGenerated = null;
+			return;
+		}
+
+		blockUI( $form );
+
+		const request = api.generatePaymentMethodRequest(
+			{
+				card: cardElement,
+			},
+			preparedCustomerData
+		);
+
+		// Populate the necessary billing details.
+		if ( useBillingDetails ) {
+			request.setBillingDetail(
+				'name',
+				(
+					$( '#billing_first_name' ).val() +
+					' ' +
+					$( '#billing_last_name' ).val()
+				).trim()
+			);
+			request.setBillingDetail( 'email', $( '#billing_email' ).val() );
+			request.setBillingDetail( 'phone', $( '#billing_phone' ).val() );
+			request.setAddressDetail( 'city', $( '#billing_city' ).val() );
+			request.setAddressDetail(
+				'country',
+				$( '#billing_country' ).val()
+			);
+			request.setAddressDetail(
+				'line1',
+				$( '#billing_address_1' ).val()
+			);
+			request.setAddressDetail(
+				'line2',
+				$( '#billing_address_2' ).val()
+			);
+			request.setAddressDetail(
+				'postal_code',
+				$( '#billing_postcode' ).val()
+			);
+			request.setAddressDetail( 'state', $( '#billing_state' ).val() );
+		}
+
+		request
+			.send()
+			.then( ( { paymentMethod } ) => {
+				// Flag that the payment method has been successfully generated so that we can allow the form
+				// submission next time.
+				paymentMethodGenerated = true;
+
+				successHandler( $form, paymentMethod );
+			} )
+			.catch( ( error ) => {
+				$form.removeClass( 'processing' ).unblock();
+				showError( error.message );
+			} );
+
+		// Prevent form submission so that we can fire it once a payment method has been generated.
+		return false;
+	};
+
+	/**
+	 * Displays the authentication modal to the user if needed.
+	 */
+	const maybeShowAuthenticationModal = () => {
+		const confirmation = api.confirmIntent( window.location.href );
+
+		// Boolean `true` means that there is nothing to confirm.
+		if ( true === confirmation ) {
+			return;
+		}
+
+		const { request, isOrderPage } = confirmation;
+
+		if ( isOrderPage ) {
+			blockUI( $( '#order_review' ) );
+			$( '#payment' ).hide( 500 );
+		}
+
+		// Cleanup the URL.
+		// https://stackoverflow.com/questions/1397329/
```

Comment:
StackOverflow has a shorthand URL that might be clearer here:

https://stackoverflow.com/a/5298684

Reply:
Thank you for the suggestion, it's implemented in https://github.com/Automattic/woocommerce-payments/pull/723/commits/70078ba71059638fa86d08b5126104b287896b20.